datashape/coretypes.py

```diff
@@ -912,7 +914,37 @@ def __repr__(self):
         return 'dshape(%s)' % strs
 
 
-class Record(CollectionPrinter, Mono):
+class RecordMeta(type):
+    @staticmethod
+    def _unpack_slice(s, idx):
+        if not isinstance(s, slice):
+            raise TypeError(
+                'invalid field specification at position %d.\n'
+                'fields must be formatted like: {name}::{type}' % idx,
+            )
+
+        name, type_ = packed = s.start, s.step
+        if name is None:
```

Comment:
can you add a check that this is an instance of `basestring`. you can import this from `compatibility` i believe